tests/dlmalloc_test.c

```diff
@@ -20,67 +20,32 @@ int main(int ac, char **av)
     char* allocations[NUM];
     for (int i = 0; i < NUM/2; i++) {
       allocations[i] = (char*)malloc((11*i)%1024 + x);
-      //printf("zz alloc: %d\n", (int)allocations[i]);
       assert(allocations[i]);
       if (i > 10 && i%4 == 1 && allocations[i-10]) {
-        //printf("zz free: %d\n", (int)allocations[i-10]);
         free(allocations[i-10]);
         allocations[i-10] = NULL;
       }
     }
     for (int i = NUM/2; i < NUM; i++) {
       allocations[i] = (char*)malloc(1024*(i+1));
-      //printf("zz alloc: %d\n", (int)allocations[i]);
       assert(allocations[i]);
       if (i > 10 && i%4 != 1 && allocations[i-10]) {
-        //printf("zz free: %d\n", (int)allocations[i-10]);
         free(allocations[i-10]);
         allocations[i-10] = NULL;
       }
     }
     char* first = allocations[0];
     for (int i = 0; i < NUM; i++) {
       if (allocations[i]) {
-        //printf("zz free: %d\n", (int)allocations[i]);
         free(allocations[i]);
       }
     }
     char *last = (char*)malloc(512); // should be identical, as we free'd it all
-    //printf("zz last: %d\n", (int)last);
     char *newer = (char*)malloc(512); // should be different
-    //printf("zz newer: %d\n", (int)newer);
-#ifndef __APPLE__
     c1 += first == last;
     c2 += first == newer;
-#else // On OSX, it's been detected that memory is not necessarily allocated linearly, so skip this check and simulate success.
-    ++c1;
-#endif
   }
-  printf("*%d,%d*\n", c1, c2);
+  EM_ASM({
+    console.log('*' + $0 + ',' + $1 + '*');
+  }, c1, c2);
```

Comment:
Can you use `emscripten_logf` rather then EM_ASM?  (its basically why I added that function).

Reply:
Ah, thanks, my concern was to avoid malloc at all costs, but it looks like that function does so (it only allocates in JS). Done.